### PR TITLE
[iprconfig] enhancements

### PR DESCRIPTION
**Pull request by jcajka**
Makes iprconfig plugin enabled on any architecture witch uses ipr+iprutils/iprconfig(not only ppc64), fixes cmd output collection witch doesn't worked  for me(nearly no output was collected) and also collects additional information from iprconfig.

### CODE REVIEW

File: sos/plugins/iprconfig.py [hunk 1]
```diff
@@ -27,16 +27,21 @@ class IprConfig(Plugin, RedHatPlugin, UbuntuPlugin, DebianPlugin):
...
     def check_enabled(self):
-        arch = self.policy().get_arch()
-        return arch == "ppc64" and is_executable("iprconfig")
+        #check if ipr is loaded
+        ipr_loaded = self.call_ext_prog('sh -c \"lsmod | grep -qw ipr\"')
+        return ipr_loaded['status'] == 0 and is_executable("iprconfig")
 
     def setup(self):
-        self.add_cmd_output([
+        self.add_cmd_outputs([
```
Comment (battlemidget) [inline sos/plugins/iprconfig.py]: This method doesn't exist any longer, needs to stay `self.add_cmd_output`

**Comment by battlemidget** [comment]
Also please follow the contributor guidelines

**Comment by jcajka** [comment]
Hm... probably older version in fedora but it doesn't work there, also wiki("How to Write a Plugin") still mentions this method:
"A simpler interface to allow several commands to be run with the same timeout and path also exists:

    def add_cmd_outputs(self, cmds, timeout=300, runat=None):
        """Run a list of programs and collect the output"""
"
Sorry, missed contributor guidelines, but they are not mention in README "Patches and pull requests" and the section is quiet verbose, so I haven't searched further.
But now I see, they are buried deeper down in small link...

If I'm looking over them correctly only sign-offs are omitted from PR, and nothing else is violating them, right?

Thanks for quick feedback!

**Comment by jcajka** [comment]
Except for pep8 violations...

**Comment by battlemidget** [comment]
Yea i think the missing sign-off was it. Also you'll want to develop plugins against master rather than whats packaged in the distro

**Comment by bmr-cymru** [comment]
Urgh sorry - my fault. I need to update teh wiki. We killed all the plural (`add_copy_specs()`, `add_cmd_outputs()` etc.) and folded them into the singular methods - they all now accept either a single string (`"string"`) or a list of strings (`[ "string1",= "string2" ]`). Sorry for the confusion.

**Comment by bmr-cymru** [comment]
>  they are not mention in README "Patches and pull requests" 

Yep - we need to add that! Thanks!

**Comment by jcajka** [comment]
@bmr-cymru :
>Shell pipelines are not allowed in command calls - the Popen call is made with shell=False. You'll >need to do this in python although to be honest I don't really see the point; what's the harm of >running iprconfig if the driver is not loaded?

It shouldn't harm. Reducing it to only check for iprutils(iprconfig) will generally enable this plugin on all systems(at least in Fedora world, if newer sos is included, as iprutils(iprconfig) are installed on all systems by default). Are you OK with this change?

> This worked fine before your patch. Now it doesn't ;-) (you need to test upstream patches with the 
> current upstream version).

Been developing against Fedora rawhide and upstream docs doesn't suggested that anything changed :). But I should have noticed... sorry.

Will fix all issues and update PR shortly.

**Comment by bmr-cymru** [comment]
> Sorry, missed contributor guidelines, but they are not mention in README "Patches and pull requests"

They are mentioned in the "Documentation" section of the file (as they are docs :-). I've added the following change to the "Patches and pull requests" section:

```diff
+Please refer to the [contributor guidelines][0] for guidance on formatting
+patches and commit messages.
+
```

**Comment by bmr-cymru** [comment]
Ping: there are a couple of minor review points here that haven't been addressed. I can do them myself but I can't guarantee to have time to do that before 3.3 is released.

**Comment by battlemidget** [comment]
@jcajka ping on this PR otherwise this may slip 3.3

**Comment by bmr-cymru** [comment]
@jcajka any chance to update this? 3.3 will be closing very soon.

**Comment by bmr-cymru** [comment]
Any updates to this PR? If not then I we'll just redo the commits and merge (although testing from someone with an IPR adapter would be preferred).

**Comment by bmr-cymru** [comment]
Rebased & merged (with style fix).

**Pull request closed without merging**